Source/WebCore/css/process-css-pseudo-selectors.py

```diff
@@ -376,63 +389,265 @@ def write_parsing_function_definitions_for_pseudo_element(self, writer):
+
+            if key_is_true(pseudo_data, 'shadow'):
+                if 'condition' in pseudo_data:
+                    shadow_map[pseudo_name] = (enablement_condition, pseudo_data['condition'])
+                else:
+                    shadow_map[pseudo_name] = enablement_condition
+
+                if 'aliases' in pseudo_data:
+                    for alias in pseudo_data['aliases']:
+                        if 'condition' in pseudo_data:
+                            shadow_alias_map[alias] = (enablement_condition, pseudo_data['condition'])
+                        else:
+                            shadow_alias_map[alias] = enablement_condition
+                continue
+
+            # Generate code for non-shadow pseudo elements.
+            if 'condition' in pseudo_data:
+                writer.write('#if {}'.format(pseudo_data['condition']))
+
+            with writer.indent():
+                writer.write('case CSSSelector::PseudoElement::{}:'.format(format_name_for_enum_class(pseudo_name)))
+
+                with writer.indent():
+                    writer.write('return {};'.format(enablement_condition))
+
+            if 'condition' in pseudo_data:
+                writer.write('#endif')
+
+        # Generate code for shadow pseudo elements.
+        def write_condition_cases_for_shadow_elements(map):
+            for pseudo_name, conditions in map.items():
+                has_ifdef_condition = type(conditions) == tuple
+                enablement_condition = conditions[0] if has_ifdef_condition else conditions
+
+                if has_ifdef_condition:
+                    writer.write('#if {}'.format(conditions[1]))
+
+                with writer.indent(), writer.indent():
+                    if ' && ' in enablement_condition:
+                        enablement_condition = '({})'.format(enablement_condition)
+
+                    writer.write('if (!{} && equalLettersIgnoringASCIICase(name, "{}"_s))'.format(enablement_condition, pseudo_name))
+                    with writer.indent():
+                        writer.write('return false;')
+
+                if has_ifdef_condition:
+                    writer.write('#endif')
+
+            with writer.indent(), writer.indent():
+                writer.write('return true;')
+
+        with writer.indent():
+            writer.write('case CSSSelector::PseudoElement::WebKitCustom:')
+        write_condition_cases_for_shadow_elements(shadow_map)
+
+        with writer.indent():
+            writer.write('case CSSSelector::PseudoElement::WebKitCustomLegacyPrefixed:')
+        write_condition_cases_for_shadow_elements(shadow_alias_map)
+
+        with writer.indent():
+            writer.write("default:")
+            with writer.indent():
+                writer.write("return true;")
+
+            # End switch statement.
+            writer.write("}")
+            writer.write("return true;")
```

Comment:
Nit: ASSERT_NOT_REACHED?

Reply:
C++ doesn't require you to return if the function only has a switch statement, so I guess I'll just omit the return, like in: https://searchfox.org/wubkat/source/__GENERATED__/WebCore/DerivedSources/CSSPropertyParsing.cpp#195-206

Reply:
It's fine because you have a `default` case, so the function is guaranteed to return and there's no point in adding an assert.

(If it were missing the `default` case, then the Linux EWS would have failed on -Werror=return-type.)